faas/event.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faas
+
+type EventHandler = func(*EventContext) (*EventContext, error)
+type EventMiddleware = func(*EventContext, EventHandler) (*EventContext, error)
+
+func eventDummy(ctx *EventContext) (*EventContext, error) {
+	return ctx, nil
+}
+
+type chainedEventMiddleware struct {
+	fun      EventMiddleware
```

Comment:
Should these be func?

Reply:
avoided `func` as it's a language keyword.